pilot/pkg/model/telemetry.go

```diff
@@ -855,9 +857,94 @@ func mergeMetrics(metrics []*tpb.Metrics, mesh *meshconfig.MeshConfig) map[strin
 		})
 		processed[provider] = tmm
 	}
+
+	processed = postProcessMetrics(processed)
 	return processed
 }
 
+// postProcessMetrics tries to minimize the metricsConfig
+func postProcessMetrics(cfg map[string]metricsConfig) map[string]metricsConfig {
+	for key, m := range cfg {
+		m.ClientMetrics = simplyMetricConfig(m.ClientMetrics)
+		m.ServerMetrics = simplyMetricConfig(m.ServerMetrics)
+		cfg[key] = m
+	}
+	return cfg
+}
+
+// maxVal equals to 2^(len(tpb.MetricSelector_IstioMetric_name)-) - 1, currently it is 1023
+var maxVal = func() int {
+	res := 0
+	for k := range tpb.MetricSelector_IstioMetric_name {
+		if k == int32(tpb.MetricSelector_ALL_METRICS) {
+			continue
+		}
+		res |= 1 << int(k-1)
+	}
+	return res
+}()
+
+func simplyMetricConfig(cfg metricConfig) metricConfig {
```

Comment:
This algorithm is not very obvious what it does. Can we add comments about what we are doing?

Reply:
updated